mypy/fastparse.py

```diff
@@ -179,33 +173,44 @@ def __init__(self,
         self.is_stub = is_stub
         self.errors = errors
 
+        # Cache of visit_X methods keyed by type of visited object
+        self.visitor_cache = {}  # type: Dict[type, Callable[[Optional[AST]], Any]]
+
     def note(self, msg: str, line: int, column: int) -> None:
         self.errors.report(line, column, msg, severity='note')
 
     def fail(self, msg: str, line: int, column: int) -> None:
         self.errors.report(line, column, msg, blocker=True)
 
-    def generic_visit(self, node: ast3.AST) -> None:
-        raise RuntimeError('AST node not implemented: ' + str(type(node)))
-
-    def visit(self, node: Optional[ast3.AST]) -> Any:  # same as in typed_ast stub
+    def visit(self, node: Optional[AST]) -> Any:
         if node is None:
             return None
-        return super().visit(node)
+        typeobj = type(node)
+        visitor = self.visitor_cache.get(typeobj)
+        if visitor is None:
+            method = 'visit_' + node.__class__.__name__
+            visitor = getattr(self, method)
+            self.visitor_cache[typeobj] = visitor
+        return visitor(node)
+
+    def set_line(self, node: N, n: Union[ast3.expr, ast3.stmt]) -> N:
```

Comment:
Just a random idea, maybe it is possible to set line at the end of `visit()` above and avoid "manually" calling it everywhere?

Reply:
It's like this because it should only be done for certain node types, not always.